src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -141,11 +109,40 @@ def uploadWorker(workInput, results, dbsUrl):
     return
 
 
+def isPassiveError(exceptionObj):
+    """
+    This function will parse the exception object and decide whether
+    the error should be dealt with in a passive way, or if it's a
+    hard error and the component better crash
+    :param exceptionObj:
+    :return: a tuple with a boolean (passive or not) and a string with
+        the error message
+    """
+    passException = True
+    passiveErrorMsg = ['Service Unavailable', 'Service Temporarily Unavailable',
+                       'Proxy Error', 'Error reading from remote server',
+                       'Connection refused', 'timed out', 'Could not resolve',
+                       'OpenSSL SSL_connect: SSL_ERROR_SYSCALL']
+
+    excReason = getattr(exceptionObj, 'reason', '')
+    errorMsg = 'Failed to fetch parentage map from WMStats, skipping this cycle. '
+    errorMsg += 'Exception: {}. Reason: {}. Error: {}. '.format(type(exceptionObj).__name__,
+                                                                excReason, str(exceptionObj))
```

Comment:
I am not quite confident it is a good idea of having this message taken out of  `DBSUploadPoller` and putting it here. I would preserve the messages related to some kind of exceptions inside the object instance of the above class, while this function here sounds to me a decent one to be considered as a generic function for testing passive vs. hard errors coming from a module. I can already foresee a place or two I could benefit from it. Putting this message here constrains its usage to only one particular type of calls and at the same time cuts significant information in the class it was taken from.

Reply:
To make it more generic, we would have to move it to a different place, and accept the list of errors as a parameter.
Moving the error message back to the class object implementation is feasible, but I'm trying to avoid code duplication; thus just profiting of some exception parsing that is already done in this function. Let me see if it makes sense to change it.

Reply:
Reading the list of errors as a parameter is not a bad idea indeed.